table/tables/tables.go

```diff
@@ -607,7 +623,13 @@ func (t *TableCommon) AddRecord(sctx sessionctx.Context, r []types.Datum, opts .
 	}
 
 	if m := t.Meta(); m.TempTableType == model.TempTableGlobal {
-		addTemporaryTable(sctx, m)
+		if tmpTable := addTemporaryTable(sctx, m); tmpTable != nil {
+			if tmpTable.GetSize() > sctx.GetSessionVars().TempTableMaxRAM {
+				// TODO: what's the standard message?
+				return nil, errors.New("temporary table size exceed @@global.temptable_max_ram")
```

Comment:
Why not resolve the TODO?

Reply:
The test is boring, I do not install a MySQL locally, and to reproduce this error message a large insert data has to be constructed.
This is just an errors message and I think it may not worth the effort now, so I just leave a TODO comment.

Reply:
Will you resolve the TODO before the GA?

Reply:
WOW, just now I verified it, I doubt does this variable really works in MySQL.

I create a normal table t1, and then fill some  data to t1:

```
 insert into t1 values (42, repeat('a', 512));
```

and then make t1 larger:

```
insert into t1 select * from t1
```
and then create the temporary table t, and insert t1 data into t:

```
mysql> insert into t select * from t1;
Query OK, 199168 rows affected (0.74 sec)
Records: 199168  Duplicates: 0  Warnings: 0
```

As you can see, the data is large then `@@global.temptable_max_ram`, but there is no error:


```
mysql> show create table t;
+-------+----------------------------------------------------------------------------------------------------------------------------------------------------------+
| Table | Create Table                                                                                                                                             |
+-------+----------------------------------------------------------------------------------------------------------------------------------------------------------+
| t     | CREATE TEMPORARY TABLE `t` (
  `id` int DEFAULT NULL,
  `c` varchar(512) DEFAULT NULL
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_0900_ai_ci |
+-------+----------------------------------------------------------------------------------------------------------------------------------------------------------+
1 row in set (0.00 sec)

mysql> select @@global.temptable_max_ram;
+----------------------------+
| @@global.temptable_max_ram |
+----------------------------+
|                    2097152 |
+----------------------------+
1 row in set (0.00 sec)

mysql> select count(*) from t;
+----------+
| count(*) |
+----------+
|  4381696 |
+----------+
1 row in set (6.19 sec)
```

You can try to reproduce it in a MySQL server @wjhuang2016

Reply:
temptable_max_ram is deprecated in MySQL 8.0.26, should we need to support it again? https://dev.mysql.com/doc/refman/8.0/en/internal-temporary-tables.html

Reply:
So shall we change this to `tmp_table_size`?
https://dev.mysql.com/doc/refman/8.0/en/server-system-variables.html#sysvar_tmp_table_size

@wjhuang2016 @djshow832

Reply:
> So shall we change this to `tmp_table_size`?
> https://dev.mysql.com/doc/refman/8.0/en/server-system-variables.html#sysvar_tmp_table_size
> 
> @wjhuang2016 @djshow832

There is a slightly behavior difference between the two settings:
- `temptable_max_ram` is the global setting that applies to the TempTable engine, and the memory buffer size is shared amongst all temporary tables on the server.
- `tmp_table_size` is a per-temporary table limit, and has session/global scope.

I assume based on our implementation that `tmp_table_size` may indeed be more accurate.